Make sure _ignoreScroll gets cleared

tomivirkki opened the pull request:
If scroller._ignoreScroll is true while resizing the grid, the headers/footers might get misplaced. This PR makes sure  _ignoreScroll flag gets cleared.
<img width="457" alt="screenshot 2016-10-07 12 00 00" src="https://cloud.githubusercontent.com/assets/1222264/19193232/f7f3f408-8cb1-11e6-92fd-f20e42e54161.png">

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-grid/481)
<!-- Reviewable:end -->

Saulis commented:
The `_ignoreScroll` is only for optimizing, right? What if we would just remove the optimization to clean up the code a bit? Didn't test on IE or Edge, but on browsers I can't tell any difference performance wise.

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-:-KThloWG6vxrjvpWMfJy:b-ns2z2q)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
It is for optimizing only. I'd assume the performance impact to be quite big though (might not be visible on desktop) as all the logic that's run on scroll would be run twice without it. If there's no notable difference on mobile devices then +1 for removing it.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-:-KThp9LPOATRm24uMwV9:b-k04bqm)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Mobile devices don't set the `ignoreScroll` to `true` as they don't fire `wheel` events right?

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-:-KTi-AyewkcPqmHd9L8M:b-4ar4wi)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
They still use the outer-scroller which does the same thing. But what I mostly meant was that we should test with something less performant than your super-laptop :)

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-:-KTi-XarTHpntdrlJ7rh:bh7mu54)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Oh right, I'd totally missed that outer scroller also uses the property.

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-:-KTiDIG69CV72VXMhjoG:b-hn5kp3)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[test/row-height.html, line 230 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-KTmBSWbGDswVUNY00L7:-KTmBSWbGDswVUNY00L8:b-xhuped) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d6934847d20dfbb33288c9168e6466dae95a23f0/test/row-height.html#L230)):*
> ```HTML
>               done();
>             }, 100);
>           }, 20); // Debounce perior for 'clear-ignorescroll' is 10
> ```

perior -> period  ?

Shouldn't  call flushDebounced instead of wait for it ? 

---

*[test/row-height.html, line 252 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-KTmBNA9zgL5snJo9Rx0:-KTmBNA9zgL5snJo9Rx1:b3rfc8w) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d6934847d20dfbb33288c9168e6466dae95a23f0/test/row-height.html#L252)):*
> ```HTML
>         });
>       });
>       
> ```

nit : some spaces left

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[test/row-height.html, line 230 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-KTmBSWbGDswVUNY00L7:-KTmOqvLz0-OUSV4vBK3:bjv50mc) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d6934847d20dfbb33288c9168e6466dae95a23f0/test/row-height.html#L230)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> perior -> period  ?
> 
> Shouldn't  call flushDebounced instead of wait for it ? 
</details>
Fixed the typo.

The timeout is needed as the async 'scroll' event might still be on the way even if we flushed the debouncer.

Also I think this test shouldn't flush the debouncer that the same PR introduces; from TDD point of view the debouncer doesn't exist (before the fix).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[test/row-height.html, line 252 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-KTmBNA9zgL5snJo9Rx0:-KTmPRmAst-n2gccfTQR:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d6934847d20dfbb33288c9168e6466dae95a23f0/test/row-height.html#L252)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> nit : some spaces left
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481)*
<!-- Sent from Reviewable.io -->

Saulis commented:
OK how about this: can we drop listening for `scroll` events altogether on the table, if we move the `wheel` listener to the grid element? Then all scrolling for the table should happen either through the `wheel` event handler on the grid or the outer scroller? That would also allow dropping the scroll syncing from table -> outer-scroller, right?

---

Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-:-KTmXbMnGoUH95O-NKGX:bxd1dul)*
<!-- Sent from Reviewable.io -->

Saulis commented:
What I kinda mean is that `wheel` events would scroll the outer-scroller instead of the table.

---

Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-:-KTmYmxMwRuyou7mLJGI:b-djkws2)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
The outer-scroller isn't in use in all cases. In some environments it's hidden on purpose.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-:-KTmZKCXav5ZAVI26_Ib:bawa9ed)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Ah, OK.  I did do some perf testing on different devices (old and new ipad we have, nexus phone, surface) using the `<fps-meter>`, there is a 1-2 FPS difference depending on the device, but nothing noticeable. We probably should figure out an automated test to benchmark the performance somehow.

---

Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-:-KTmiPd-OAWAD6qIn6Ys:b-7p0nfr)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Ok, sounds good. Changed the PR so that it removes _ignoreScroll optimization.

---

Review status: 0 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[test/row-height.html, line 230 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-KTmBSWbGDswVUNY00L7:-KTmumUFkS4bNgZgx2NU:b-1urd85) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d6934847d20dfbb33288c9168e6466dae95a23f0/test/row-height.html#L230)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary>
> Fixed the typo.
> 
> The timeout is needed as the async 'scroll' event might still be on the way even if we flushed the debouncer.
> 
> Also I think this test shouldn't flush the debouncer that the same PR introduces; from TDD point of view the debouncer doesn't exist (before the fix).
</details>
Fixed, the flag was removed so no need to wait > the debounce period

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-:-KTmui6LqGryzamctzgG:bs3ysfv)*
<!-- Sent from Reviewable.io -->

Saulis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 3 files at r1, 5 of 5 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/481#-:-KTnpt-eOI7IssL1Qq2X:bnfp4nl)*
<!-- Sent from Reviewable.io -->
